question-service/index.js

```diff
@@ -24,7 +24,21 @@ app.get('/questions/:difficulty', (req, res) => {
   const questionArr = processedQuestions[difficulty];
 
   questionObj = questionArr?.[Math.floor(Math.random() * questionArr.length)];
-  
+
+  if (!questionObj) {
+    return res.status(404).json({ err: "No question found!" })
+  }
+
+  res.status(200).json({ message: "Question retrieved!", data: questionObj });
+});
+
+app.get('/questions/:difficulty/:id', (req, res) => {
+  const { difficulty, id } = req.params;
+  let questionObj;
+  const questionArr = processedQuestions[difficulty];
+
+  questionObj = questionArr?.find((question) => question.id.toString() === id);
```

Comment:
can `questionObj` be a `const` here ? defined after `questionArr` of course.

Reply:
sure. I simply followed the previous route for the same format. Both will be updated